client/components/jetpack/intro-pricing-banner-v2/index.tsx

```diff
@@ -0,0 +1,89 @@
+import { useViewportMatch } from '@wordpress/compose';
+import { useTranslate } from 'i18n-calypso';
+import { FunctionComponent, useMemo } from 'react';
+import useDetectWindowBoundary from 'calypso/lib/detect-window-boundary';
+import { preventWidows } from 'calypso/lib/formatting';
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import { INTRO_PRICING_DISCOUNT_PERCENTAGE } from 'calypso/my-sites/plans/jetpack-plans/constants';
+import { isConnectStore } from 'calypso/my-sites/plans/jetpack-plans/product-grid/utils';
+import { JetpackSaleCoupon } from 'calypso/state/marketing/selectors';
+import './style.scss';
+import guaranteeBadge from './14-day-badge.svg';
+import rocket from './rocket.svg';
+
+// since this amount is backed into the badge above we make it a const
+const GUARANTEE_DAYS = 14;
+
+interface Props {
+	jetpackSaleCoupon?: JetpackSaleCoupon | null;
+}
+
+const IntroPricingBannerV2: FunctionComponent< Props > = ( { jetpackSaleCoupon = null } ) => {
+	const translate = useTranslate();
+	const isNotNarrow = useViewportMatch( 'medium', '>=' );
+
+	const CALYPSO_MASTERBAR_HEIGHT = 47;
+	const CLOUD_MASTERBAR_HEIGHT = 0;
+
+	const windowBoundaryOffset = useMemo( () => {
+		if ( isJetpackCloud() || isConnectStore() ) {
+			return CLOUD_MASTERBAR_HEIGHT;
+		}
+
+		return CALYPSO_MASTERBAR_HEIGHT;
+	}, [] );
+	const [ barRef, hasCrossed ] = useDetectWindowBoundary( windowBoundaryOffset );
+
+	const outerDivProps = barRef ? { ref: barRef as React.RefObject< HTMLDivElement > } : {};
+
+	const discountPercentage =
+		jetpackSaleCoupon !== null ? jetpackSaleCoupon.discount : INTRO_PRICING_DISCOUNT_PERCENTAGE;
+
+	return (
+		<>
+			<div className="intro-pricing-banner-v2__viewport-sentinel" { ...outerDivProps }></div>
```

Comment:
There's probably a few different ways to fix the 'jumping' effect, but one quick way would be to: Instead of having this 'sentinel' div on it's own, you could instead use the sentinel div as a wrapper to wrap the all the pricing banner contents, and then with CSS give it a height of 32px (desktop) & 24px (mobile).  So then when `intro-pricing-banner-v2__sticky` becomes fixed and drops out of the document flow, the sentinel div will still maintain the height of the banner, preventing/fixing the 'jumping' effect, like so:

![Screen Capture on 2022-01-11 at 10-28-33](https://user-images.githubusercontent.com/11078128/148976750-7106a0f0-e4ad-48cb-b69e-824231431a02.gif)

Reply:
I ended up moving the intro banner to the same place that the annual/monthly toggle was being rendered from. I ruthlessly ripped off that component, so putting the new banner in the same place gives us the smooth transitions.